### ISSUE TITLE
"Missing executable" on Steam

### ISSUE THREAD

**Issue by vpltd-peterm** [virtual-programming/overlord-linux#1]
We're aware that some people are getting this message.

If you have made a new purchase from today on Steam, please indicate that here.
If you have made a previous purchase, please use Account Details and check how you acquired it. It may have been part of a package bundle that we are currently unable to modify, but have asked Codemasters to look at.

It is also possible that the content has not spread to all of Steam's download servers yet. In this case, try switching Download Region in Steam

**Comment by rcpoison** [comment]
Bought today:
http://store.steampowered.com/sub/2464/
Region Europe/Germany.
Missing executables for all three.

**Comment by vpltd-peterm** [comment]
I've just updated a whole bunch of sales packages for the Overlord games that were overlooked at first. Hopefully this should largely resolve the issues being seen.

**Comment by dubigrasu** [comment]
Could you please look into these two Codemasters bundles?
https://steamdb.info/search/?a=sub&q=Codemasters%20Bundle%201
https://steamdb.info/search/?a=sub&q=Codemasters%20Bundle%202

**Comment by vpltd-peterm** [comment]
We have asked Codemasters to look at those, as only they can update them.

**Comment by Kwaadpepper** [comment]
Hey there, i am experiencing something weird with missing files, i managed to play the game pretty far, but since i had some issues, i quitted the game and revalidated it, and steam continues to say there are 74 missing files. Also text is now gone on menus.

**Comment by vpltd-peterm** [comment]
@Kwaadpepper, this is not related to the issue topic. Please make a new one to cover this issue.

**Comment by JustHev** [comment]
I've sent an email regarding this issue to Codemasters' Customer Service, this is what they replied:

> Thank you for submitting your recent request to Codemasters Customer Services concerning Overlord.
> 
> Please contact Virtual Programming regarding the Mac or Linux version of the game:
> 
> https://www.vpltd.com/
> 
> Codemasters are only the developers and publishers of the Windows game, not the Mac or Linux version, I'm afraid.
> 
> Kind regards
> Rob
> Codemasters Customer Services

Any update on this issue? It still affects me. (Owner of the Humble Weekly Codemasters Bundle)

**Comment by vpltd-peterm** [comment]
Codemasters have advised us that bundles that were available to purchase prior to the release of the Mac and Linux version do not qualify for a free copy of the game for the new platforms. These special bundles were sold purely for the supported platform at that time.

**Comment by JustHev** [comment]
This is an odd move for Codemasters to make, for several reasons:

1. SteamPlay has been popular for a few years now: most companies that release ports for OS X/Linux also give access to their previous customers. This includes games like the Saints Row series, Payday 2 and The Stanley Parable, all of which weren't advertised as being Linux compatible when I bought them. (I previously used Windows, that is why I bought these games back then). Customers that bought one or more Overlord games on Steam previously also gained access to the respective Linux versions as stated in the [announcement message](http://steamcommunity.com/games/11450/announcements/detail/928124261817146458) 

>Current owners of Overlord on Steam have immediate access to the game from SteamOS and Mac or Linux devices running Steam. 

A similar message is in the announcements for [Overlord Raising Hell](http://steamcommunity.com/games/12710/announcements/detail/887591865177093875) and [Overlord II](http://steamcommunity.com/games/12810/announcements/detail/818911970857628400). These sentences suggest that owners of the Codemasters Humble Bundle, who are also owners of Overlord, also have access to the Linux version.

2. A different precedent has been set last year, when DiRT Showdown was ported to Linux by you guys. This game was also in the same Codemasters bundle. In this bundle, Linux compatibility for DiRT Showdown hasn't been advertised as well, but owners of the Codemasters bundle gained access to the Linux version regardless, [as indicated by the Steam depot info, as indexed by SteamDB](https://steamdb.info/sub/37052/depots/). Codemasters denying access to the Linux versions of the Overlord Series is therefore contradictory with what they've done previously.

3. Even if I want to buy the game again, I can't: When I try and checkout the Overlord series on Steam, I get the following [error message](http://i.imgur.com/u3l5gLl.png): `Your account already owns some of the above items, so you can’t purchase them again. You can purchase these items as a gift or adjust the contents of your cart to continue.` Therefore I can't even *buy* the Linux versions of the Overlord series. By restricting the access to the Linux version of Overlord, Codemasters can't gain additional sales. The games from the Overlord series also [show up in my library](http://i.imgur.com/0Ku88Zr.png) as installable games, even though the games are not playable, since I don't have access to the Linux depots, which I find really annoying.

In conclusion, the Steam announcements contradict reality by stating that all Overlord owners gain access to the Linux versions, Codemasters denying access to the Linux version because it wasn't advertised with a Linux version is contradictory to what they've done with DiRT Showdown and Codemasters has no commercial interest in denying access to the previous owners, since they cannot sell the game to these customers again.

**Comment by dubigrasu** [comment]
Thank you vpltd-peterm for the info.
But JustHev is right, that's a weird thing to do for Codemasters, why we have access to Dirt Showdown but not Overlord **within the same bundle**?
I'm thinking, any point to take this to Steam support?

**Comment by dubigrasu** [comment]
Sent an email today to Codemaster about this, they promptly replied
:
>Please contact Virtual Programming regarding the Mac or Linux version of the game:
https://www.vpltd.com/
Codemasters are only the developers and publishers of the Windows game, not the Mac or Linux version, I'm afraid.

So VP says is up to Codemasters while Codemasteres says is up to VP...what now?

**Comment by deresh** [comment]
@dubigrasu:

https://github.com/virtual-programming/overlord2-linux/issues/1#issuecomment-239418348

**Comment by vpltd-plesurf** [comment]
I have spoken to a Codemasters again, informing them that it is not possible to purchase the Mac/Linux versions if the game is owned in a Windows bundle.

This has been accepted as a problem and Codemasters will now be updating all bundles to allow Mac and Linux versions.

Please give this a few days to be sorted, but it is now in motion.

Regards.

**Comment by JustHev** [comment]
@vpltd-plesurf Thank you for adressing this problem with Codemasters. I'm looking forward to finally playing the Overlord games on Linux.

**Comment by LiamDawe** [comment]
Fantastic news for all affected, great work seeing this resolved VP :+1:

**Comment by deresh** [comment]
well, this still isn't resolved for me. Are you guys @LiamDawe @JustHev having more luck?

**Comment by sgdw** [comment]
First of all, thank you for taking care of this problem. But it isn't resolved for me either.

Having a look into the local appmanifest for Overloard ID:11450, I can only see ID:11453 as a mounted depot, which is exactly the 135Mb "Overlord English" download. No executable, only some audio files.

**Comment by Seegras** [comment]
Not resolved here either. All I got was the 135MB language-data for Overlord and Overlord:Rising Hell, and none for Overlord II. I got them with "Codemasters Bundle 1" (https://steamdb.info/sub/37051/apps/) and "Codemasters Bundle 2" (https://steamdb.info/sub/37052/apps/). Download location was Germany, Frankfurt. Is this resolved in any download-location?

**Comment by JustHev** [comment]
@Seegras: No.

Last thing we know is that things have been set in motion, then the communication ended from VP. Codemasters prefers to commicate through/shift the blame to VP.

**Comment by soldari** [comment]
I am also the owner of the Humble Codemasters Bundle and while I can enjoy playing Dirt Showdown I have the same problem as Seegras and others here that Overlord, Overlord : Raising Hell and Overlord II are visible in my library yet not possible to play due to  "missing executable" error.
I understand what has happened but it is now close on a month since release and still no joy. I'd purchase an extra copy or  2 to give away on gamingonlinuxdotcom if this could just be finally sorted out.

Thanks in anticipation

**Comment by vpltd-peterm** [comment]
Codemasters still have not updated "Codemasters Bundle 1" (37051) or "Codemasters Bundle 2" (37052). We will remind them, but that's all we can do at this stage.

**Comment by dubigrasu** [comment]
Looks like is happening right now, the depots were updated and the games are receiving updates.

**Comment by deresh** [comment]
It works now. Game is sucestully downloaded and i can play it.

Thank you guys for your support!

**Comment by soldari** [comment]
Updates also downloaded today, started Overlord game and everything seems good, can finally get some time into these games.
Much thanks vpltd-peterm and team.

Will make good on my word to try show my support for your great porting efforts :)
